### PR TITLE
various: fix logging to be great again

### PR DESCRIPTION
**Pull request by michaelld**
[Lots of words below for so few changes! - MLD]

1: change logger_ptr when not ENABLE_GR_LOG to a pointer, to match the type use for the other typedefs. Addresses issue https://github.com/gnuradio/gnuradio/issues/1383 .

2: Fix FindLog4Cpp to (a) allow external setting of LOG4CPP variables, (b) use HINTS with potential provided variables, (c) correctly set cached variables to be what we expect them to be, no matter what, and (d) do LOG4CPP found variable printout only wnen LOG4CPP is enabled.

3: Reduce HAVE_LOG4CPP and LOG4CPP_FOUND down to just the latter, except when CMake has a #cmakedefine to replace.

4: Only use LOG4CPP_INCLUDES and LOG4CPP_LIBRARIES when actually required, and especially not LOG4CPP_INCLUDES in top-level variables such as GNURADIO_RUNTIME_INCLUDE_DIRS, which broken heading finding since already-installed header could be found before project ones.

5: in c++ when using a logger function, use the absolute namespace.

6: a little whitespace here and there.

### CODE REVIEW

File: gnuradio-runtime/CMakeLists.txt [hunk 1]
```diff
@@ -124,14 +123,14 @@ install(
     COMPONENT "runtime_runtime"
 )
 
-if(ENABLE_GR_LOG AND HAVE_LOG4CPP)
+if(ENABLE_GR_LOG AND LOG4CPP_FOUND)
 install(FILES
```
Comment (mbr0wn) [inline gnuradio-runtime/CMakeLists.txt]: Maybe indent this?

Comment (michaelld) [inline gnuradio-runtime/CMakeLists.txt]: It wasn't indented before but it's a short section so I'll include indenting it in the next push I do for this PR. It's not quite "there" yet if you followed the discussion. I'm almost there with overloading CMake's default `include_directories` ... very cool to be able to make a 'smart' version.

**Comment by jmcorgan** [comment]
See compile issue on buildbot:
https://ci.gnuradio.org/buildbot/#/builders/6/builds/154/steps/4/logs/warnings__1469_

**Comment by jmcorgan** [comment]
@michaelld Can you give me an ETA on addressing this?

**Comment by michaelld** [comment]
Per suggestion in https://github.com/gnuradio/gnuradio/issues/1383, I'm going to split off the logger_ptr part of this PR into a new PR. I'll create an overloaded `include_directories` macro & then tweak the use of it. Plenty of small changes, but I think we can fix the include ordering once and for all with this forthcoming change!

**Comment by jmcorgan** [comment]
Let me know the status of this now that we have #1438 and the recent include macro merged.

**Comment by michaelld** [comment]
This PR will require a little re-working now with those fixes in place. I'll redo & push the changes here & ping you when they are ready to go.

**Comment by jmcorgan** [comment]
I'm going to close this pending your rework, which can be done in a new PR off the latest master.

**Comment by michaelld** [comment]
works for me. not enough time in the day!

**Pull request closed without merging**